clifford/_multivector.py

```diff
@@ -651,33 +652,45 @@ def isScalar(self) -> bool:
 
         return True
 
-    def isBlade(self) -> bool:
+    def isBlade(self, *, invertible=False) -> bool:
         """Returns true if multivector is a blade.
         """
         if len(self.grades()) != 1:
             return False
 
-        return self.isVersor()
+        return self.isVersor(invertible=invertible)
 
-    def isVersor(self) -> bool:
+    def isVersor(self, *, invertible=True) -> bool:
         """Returns true if multivector is a versor.
         From Leo Dorsts GA for computer science section 21.5, definition from 7.6.4
         """
+        if invertible:
+            grade_cmp = operator.eq
+        else:
+            grade_cmp = operator.le  # subset
+
         Vhat = self.gradeInvol()
         Vrev = ~self
-        Vinv = Vrev/(self*Vrev)[0]
+        mag = (self*Vrev)[()]
+        if mag != 0:
+            # not strictly necessary, just scales to make eps appropriate below
+            Vinv = Vrev / mag
+        elif invertible:
+            return False
+        else:
+            Vinv = Vrev
```

Comment:
Changes here are to:

* bail out early without a divide-by-zero if not allowing non-invertible blades
* avoiding the division if it is zero, since that shouldn't affect the results below significantly
* Use `[()]` instead of `[0]`, which is a little more reliable for pulling out the scalar part